File: Where/Where/iOS/ViewController.m
//
//  ViewController.m
//  Where
//
//  Created by Nero on 10/8/15.
//  Copyright (c) 2015 Nero. All rights reserved.
//

#import "ViewController.h"
#import "MapManager.h"
#import "SearchManager.h"
#import "QuickUI.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
}

-(void) viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    
    //启动高德地图
    [[MapManager sharedInstance] initManagerSize:self.view.bounds];
    
    [[MapManager sharedInstance] showTraffic:YES];
    [[MapManager sharedInstance] setMapType:TYPE_DRAW];
    [[MapManager sharedInstance] followUserLocation:YES];
    
    [self.view addSubview:[MapManager sharedInstance].mapView];
    
    //启动地图数据搜索服务
    [[SearchManager sharedInstance] initSearchService];
    
    //添加ui层
    QuickUI* view = [QuickUI addViewWithFrame:CGRectMake(0,0,CGRectGetWidth(self.view.bounds),CGRectGetHeight(self.view.bounds))];
    [self.view addSubview:view];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Where?/Where?/System/ViewController.swift
//
//  ViewController.swift
//  Where?
//
//  Created by Nero on 10/10/15.
//  Copyright (c) 2015 Nero. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        //初始化地图view，并启用高德地图服务
        MapManager.sharedInstance().initMapServices(view.bounds)
        MapManager.sharedInstance().showTraffic = true
        MapManager.sharedInstance().showUserLocation(true)
        //初始化地理信息搜索API
        SearchManager.sharedInstance().initSearchService()
        
        view.addSubview(MapManager.sharedInstance().mapView!)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

